src/libraries/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Unary/ForAllOperator.cs

```diff
@@ -147,7 +147,7 @@ internal ForAllEnumerator(QueryOperatorEnumerator<TInput, TKey> source, Action<T
             // element action for each element.
             //
 
-            internal override bool MoveNext([MaybeNull, AllowNull] ref TInput currentElement, ref int currentKey)
+            internal override bool MoveNext([MaybeNullWhen(false), AllowNull] ref TInput currentElement, ref int currentKey)
```

Comment:
Is this correct?  With the `[AllowNull]`, someone could pass in a null `TInput` even if `TInput` is declared to be non-nullable, in which case if the method doesn't touch `currentElement`, it'll still be null on the way out.  Doesn't that mean it should remain `[MaybeNull]`?

Reply:
`QueryOperatorEnumerator` declares `internal abstract bool MoveNext([MaybeNullWhen(false), AllowNull] ref TElement currentElement, ref TKey currentKey);`.

So either `QueryOperatorEnumerator.MoveNext` needs to change to `[MaybeNull]` or `ForAllEnumerator.MoveNext` does ;-)

Reply:
This is all internal in code that hasn't changed in 10 years.  I have a fairly low care-level for the annotations; do whichever you think makes the most sense, as long as the generated IL for the actual method code doesn't change :)